mlflow/utils/validation.py

```diff
@@ -3,11 +3,12 @@
 """
 import re
 
-# Regex for metric and parameter names: we only allow [a-zA-Z0-9_-] ([\w-]), / and . characters,
+# Regex for metric and parameter names: we only allow [a-zA-Z0-9_- ] ([\w-]), / and . characters,
 # and we do not allow adjacent, leading or trailing / or path components that are empty or only
 # contain dots. The goal is to make our names be valid, unique path names too in order to treat
 # them as such in the UI and file store.
-_METRIC_AND_PARAM_NAME_REGEX = re.compile(r"^([\w.-]*[\w-][\w.-]*)(/[\w.-]*[\w-][\w.-]*)*$")
+_METRIC_AND_PARAM_NAME_REGEX = re.compile(
+    r"^([\w.\- ]*[\w\-][\w.\- ]*)(/[\w.\- ]*[\w\-][\w.\- ]*)*$")
```

Comment:
If you want to allow spaces everywhere, shouldn't the middle `[\w\-]` be `[\w\- ]`? Or do you want to disallow metrics whose names are just spaces? That's OK but it will just be harder to describe to people.

Reply:
Fine with allowing all spaces everywhere. I was going for a "there needs to be at least some other character", but we don't need to be so prescriptive.

Reply:
OK, I guess this is fine as is. BTW we should also add this to the docs somewhere so people can look up what valid metric and parameter names are.